server/plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/ui/GLSPDiagramEditor.java

```diff
@@ -372,28 +372,26 @@ protected void installBrowserFunctions() {
       ChromiumSelectionFunction.install(GLSPDiagramEditor.this, browser);
    }
 
+   protected String getBaseUrl() { return "diagram.html"; }
+
    protected String createBrowserUrl() {
-      try {
-         String path = getFilePath();
-         GLSPServerManager manager = getServerManager();
-         ServerConnector connector = Stream.of(manager.getServer().getConnectors()).findFirst()
-            .map(ServerConnector.class::cast)
-            .orElse(null);
-
-         if (connector != null) {
-            String url = String.format("http://%s:%s/diagram.html?client=%s&path=%s&port=%s&widget=%s",
-               connector.getHost(),
-               manager.getLocalPort(),
-               encodeParameter(getClientId()),
-               encodeParameter(path),
-               manager.getLocalPort(),
-               encodeParameter(getWidgetId()));
-            System.out.println(url);
-            return url;
-         }
-      } catch (UnsupportedEncodingException exception) {
-         LOGGER.error(exception);
+      String path = getFilePath();
+      GLSPServerManager manager = getServerManager();
+      ServerConnector connector = Stream.of(manager.getServer().getConnectors()).findFirst()
+         .map(ServerConnector.class::cast)
+         .orElse(null);
+
+      if (connector != null) {
+         Map<String, String> queryParams = new HashMap<>();
+         queryParams.put("client", clientId);
+         queryParams.put("path", path);
+         queryParams.put("port", "" + manager.getLocalPort());
+         queryParams.put("widget", getWidgetId());
+         queryParams.put("application", APPLICATION_ID);
```

Comment:
Maybe we can have a dedicated protected method to overwrite/add additional parameters without having to override the complete method?